### ISSUE TITLE
about response header name repeat

### ISSUE THREAD

**Issue by shirakun** [ixudra/curl#108]
Hi

I found out about a header returning information containing duplicate name

> Request  Code

```
        $response = Curl::to("http://xxx.com/?do=login")
            ->withHeaders([
                "Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3",
                "Accept-Language: zh-TW,zh-HK;q=0.9,zh-CN;q=0.8,zh;q=0.7,ja-JP;q=0.6,ja;q=0.5,en-US;q=0.4,en-HK;q=0.3,en;q=0.2,ru;q=0.1,fa;q=0.1,fr;q=0.1",
                'Cookie: ' . $this->session,
                'User-Agent: ' . $this->ua,
                'Referer: http://www.xxx.com/',
            ])
            ->withData([
                'nick' => "123",
                'pwd'  => "456",
                'ft'   => 1,
                'code' => 7105,
            ])
            ->withResponseHeaders()
            ->returnResponseObject()
            ->post();
        var_dump($response->headers);

```

> dump

```
array(12) {
   ["Server"]=>
   string(6) "Apache"
   ["Connection"]=>
   string(10) "keep-alive"
   ["Date"]=>
   string(29) "Thu, 12 Sep 2019 02:55:34 GMT"
   ["Cache-Control"]=>
   string(25) "must-revalidate, no-store"
   ["Content-Type"]=>
   string(24) "text/html; charset=UTF-8"
   ["Content-Length"]=>
   string(2) "41"
   ["X-NWS-UUID-VERIFY"]=>
   string(32) "xxxxxxx"
   ["Pragma"]=>
   string(8) "no-cache"
   ["Set-Cookie"]=>
   string(14) "pwd=xxx"
   ["Vary"]=>
   string(10) "User-Agent"
   ["X-Daa-Tunnel"]=>
   string(11) "hop_count=3"
   ["X-NWS-LOG-UUID"]=>
   string(53) "xxx xxx"
   }
```
    

> postman return header

![image](https://user-images.githubusercontent.com/11274807/64750410-68dba780-d54b-11e9-8e81-36fffdc59cc1.png)

The correct request returns three Set-Cookies
But at this point I can only get a Set-Cookie value, because there can't be duplicate keys in the array.
Is there any other way to solve this problem?

**Comment by elimentz** [comment]
Not sure, I think I'll have to fix in code. To clarify: the `$this->session` is an array and only of the the values instead of all of them is included in the request, correct?

**Comment by shirakun** [comment]
> Not sure, I think I'll have to fix in code. To clarify: the `$this->session` is an array and only of the the values instead of all of them is included in the request, correct?

`$this->session` is a string that is used for requests, which is not the key to the problem.

The header in response returns multiple `Set-Cookies'(as shown in the figure in the question)

However, `$response->headers` get an array

As you know, keys in arrays don't duplicate, that is, three `Set-Cookies` only keep the last one.

----------------------------

Let's give a simple example.

```
var_dump([
"Set-Cookie"=>"value1",
"Set-Cookie"=>"value2",
"Set-Cookie"=>"value3",
]);
```
It will never output

```
array(3) {
["Set-Cookie"]=>
string(6) "value1"
["Set-Cookie"]=>
string(6) "value2"
["Set-Cookie"]=>
string(6) "value3"
}

```

**Comment by elimentz** [comment]
Ah, so the request is fine but the return headers are the problem?

**Comment by elimentz** [comment]
Gotcha, that's definitely something I need to look at in code. Won't be till the weekend though, I have a lot on my plate right now. If you want to speed things up, feel free to make a PR

**Comment by shirakun** [comment]
> Gotcha, that's definitely something I need to look at in code. Won't be till the weekend though, I have a lot on my plate right now. If you want to speed things up, feel free to make a PR

Okey, I'll try to solve it, but I don't necessarily make a PR, because my coding level is not very good.

**Comment by elimentz** [comment]
No worries. Together, we'll get there. If you do find a solution, be sure to let me know. I'll take care of putting it into the package.

**Comment by elimentz** [comment]
Fixed in version 6.17.0! Thanks for the contribution!
